js/webui/src/mediasession_controller.js

```diff
@@ -0,0 +1,78 @@
+import DataSource from "./data_source";
+import PlayerModel from "./player_model";
+import silenceMp3 from "./5-seconds-of-silence.mp3";
+
+export default class MediaSessionController {
+  /**
+   * @param {any} client
+   * @param {PlayerModel} playerModel
+   */
+  constructor(client, playerModel) {
+    this.dataSource = new DataSource(client);
+    this.playerModel = playerModel;
+  }
+
+  start() {
+    if (!"mediaSession" in navigator) {
+      return;
+    }
+
+    navigator.mediaSession.setActionHandler("play", () =>
+      this.playerModel.play()
+    );
+    navigator.mediaSession.setActionHandler("pause", () =>
+      this.playerModel.pause()
+    );
+    navigator.mediaSession.setActionHandler("previoustrack", () =>
+      this.playerModel.previous()
+    );
+    navigator.mediaSession.setActionHandler("nexttrack", () =>
+      this.playerModel.next()
+    );
+
+    this.dataSource.on("player", (player) => this.updateMetadata(player));
+    this.dataSource.watch("player", {

```

Comment:
Only one type can watch on data source, this logic should be PlayerModel instead.

Reply:
Moved to PlayerModel, but it should probably be refactored later because the activeItem's columns are becoming messy